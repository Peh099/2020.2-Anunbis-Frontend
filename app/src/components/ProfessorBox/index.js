import React from 'react';
import Post from '../Post';

import Button from '../Button';
import { ProfessorBoxStyle, Info_ProfessorBox, Posts_ProfessorBox, HeaderStyle } from './styles.js'

const Info_Teacher = ({ children }) => {
  return (
    <Info_ProfessorBox>
      <span>
        {children[0]}
      </span>
      <span>
        Nota: {children[1]?.toFixed(2)}
      </span>
    </Info_ProfessorBox>
  );
}

const Posts = ({ children, report}) => {

  return (
    <Posts_ProfessorBox>
      {children?.map(post => {
        return (
          <span>
            <Post report={report} post={post} />
          </span>)
      })}
    </Posts_ProfessorBox>
  );
}

export default function ProfessorBox({onClick, report, name, rating, posts}) {
  
  return (
    <ProfessorBoxStyle>
      <HeaderStyle>
        <Info_Teacher>
          {name}
          {rating}
        </Info_Teacher>
        <Button text="AVALIAR" onClick={onClick}/>
      </HeaderStyle>
      <Posts 
        report={report} 
        >
          {posts}
      </Posts>
    </ProfessorBoxStyle>
  );
}