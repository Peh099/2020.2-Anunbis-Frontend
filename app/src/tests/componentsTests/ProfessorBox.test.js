import renderer from 'react-test-renderer';
import ProfessorBox from '../../components/ProfessorBox';

describe('Snapshot ProfessorBox component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<ProfessorBox />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})