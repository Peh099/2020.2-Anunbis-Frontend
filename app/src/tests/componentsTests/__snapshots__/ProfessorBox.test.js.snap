// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ProfessorBox component matches the snapshot 1`] = `
<div
  className="sc-hHEiqL cRfAQS"
>
  <header
    className="sc-fKgJPI ctsYtX"
  >
    <div
      className="sc-dlMDgC eDxiqV"
    >
      <span />
      <span>
        Nota: 
      </span>
    </div>
    <button
      className="sc-bdnxRM bSShnC"
    >
      AVALIAR
    </button>
  </header>
  <div
    className="sc-kfYoZR eaqZXy"
  />
</div>
`;
